### PR TITLE
solution

### PR DESCRIPTION
**Pull request by tarasulo**

### CODE REVIEW

File: consumer/src/main/resources/hibernate.cfg.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kolyansv) [inline consumer/src/main/resources/hibernate.cfg.xml]: do you need this file at all?

File: consumer/src/main/java/dao/AbstractDao.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package dao;
+
+import java.sql.Connection;
+
+public class AbstractDao<T> {
+    protected final Connection connection;
```
Comment (kolyansv) [inline consumer/src/main/java/dao/AbstractDao.java:6]: when do you close db connection?

File: consumer/src/main/java/dao/CarDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import model.Car;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public class CarDaoJdbcImpl extends AbstractDao<Car> {
```
Comment (kolyansv) [inline consumer/src/main/java/dao/CarDaoJdbcImpl.java:13]: what is the reason of extending abstract class?

File: consumer/src/main/java/dao/CarDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            + "(brand, model, year, engine) VALUES (?, ?, ?, ?);";
+
+    private final static Logger logger = LoggerFactory.getLogger(CarDaoJdbcImpl.class);
+
+    public CarDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    public Car add(Car finalCar) {
+        try (PreparedStatement statement = connection.prepareStatement(
```
Comment (kolyansv) [inline consumer/src/main/java/dao/CarDaoJdbcImpl.java:26]: btw, where do you close prepared statement?

File: consumer/src/main/java/dao/CarDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+                queryAddCar, PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, finalCar.getBrand());
+            statement.setString(2, finalCar.getModel());
+            statement.setInt(3, finalCar.getYear());
+            statement.setDouble(4, finalCar.getEngine());
+            statement.executeUpdate();
```
Comment (kolyansv) [inline consumer/src/main/java/dao/CarDaoJdbcImpl.java:32]: better to print number of affected records

File: filter/src/main/java/controller/MessageFilterController.java [hunk 6]
```diff
@@ -49,35 +46,30 @@ public static void main(String[] args) {
...
-                            producer.close();
-                        } catch (Exception e) {
-                            logger.error("Resend failed " + String.valueOf(e));
-                        }
-                    } else {
-                        logger.info("Sorry, these " + tempCar.toString() + ", does not meet the filtering requirements");
-                    }
-
+                } else {
+                    logger.info("Sorry, these " + tempCar.toString() + ", does not meet the filtering requirements");
```
Comment (kolyansv) [inline filter/src/main/java/controller/MessageFilterController.java]: do not use concatenation if you are using logback. Check how it be done properly and update all logs.

File: consumer/src/main/resources/hibernate.cfg.xml (hunk 1 above)
Comment (tarasulo) [reply consumer/src/main/resources/hibernate.cfg.xml]: No, we used JDBC

File: consumer/src/main/java/dao/CarDaoJdbcImpl.java (hunk 3 above)
Comment (tarasulo) [reply consumer/src/main/java/dao/CarDaoJdbcImpl.java:13]: it makes sense when we have several different objects

File: filter/src/main/java/controller/MessageFilterController.java (hunk 6 above)
Comment (tarasulo) [reply filter/src/main/java/controller/MessageFilterController.java]: done

File: consumer/src/main/java/dao/CarDaoJdbcImpl.java (hunk 4 above)
Comment (tarasulo) [reply consumer/src/main/java/dao/CarDaoJdbcImpl.java:26]: it's closed automatically. when I'm writing statement.close(); - Idea told me "remove redundant close()"

File: consumer/src/main/java/dao/AbstractDao.java (hunk 2 above)
Comment (tarasulo) [reply consumer/src/main/java/dao/AbstractDao.java:6]: done

File: consumer/src/main/java/dao/CarDaoJdbcImpl.java (hunk 5 above)
Comment (tarasulo) [reply consumer/src/main/java/dao/CarDaoJdbcImpl.java:32]: add id to object in the database

File: server/src/main/java/server/Server.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    private static CarService carService = new CarService();
+    private ServerSocket ss = null;
+
+    public void runServer() {
+        try {
+            ss = new ServerSocket(port);
+            logger.info("System is ready to accept the connection");
+            while (true) {
+                Socket socket = ss.accept();
+                ObjectInputStream is = new ObjectInputStream(socket.getInputStream());
```
Comment (kolyansv) [inline server/src/main/java/server/Server.java]: why do you need input stream here?

File: server/src/main/java/server/Server.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+                Car car = (Car) is.readObject();
```
Comment (kolyansv) [inline server/src/main/java/server/Server.java]: Why does your server need to read car at all? It should generate new car.
Could you please explain this?

File: server/src/main/java/server/Server.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class Server {
+
+    private final static Logger logger = LoggerFactory.getLogger(Server.class);
+    public static final int port = 4444;
```
Comment (kolyansv) [inline server/src/main/java/server/Server.java]: do NOT hardcode port

File: server/src/main/java/server/Server.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    private static CarService carService = new CarService();
```
Comment (kolyansv) [inline server/src/main/java/server/Server.java]: why do you need CarService as static variable?

File: server/src/main/java/server/Server.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    private ServerSocket ss = null;
+
+    public void runServer() {
+        try {
+            ss = new ServerSocket(port);
+            logger.info("System is ready to accept the connection");
+            while (true) {
+                Socket socket = ss.accept();
```
Comment (kolyansv) [inline server/src/main/java/server/Server.java]: do you really need to accept same client in the while(true) loop?

File: server/src/main/java/server/Server.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class Server {
+
+    private final static Logger logger = LoggerFactory.getLogger(Server.class);
```
Comment (kolyansv) [inline server/src/main/java/server/Server.java]: what is convention for constants in java?

File: producer/src/main/java/controller/NewCarsSender.java [hunk 13]
```diff
@@ -35,13 +36,17 @@ public static void main(String[] args) throws InterruptedException {
         Producer<String, Car> producer = new KafkaProducer<String, Car>(props);
 
         while (true) {
-            //Creating new car
             Car car = new Car();
-            carService.makeCar(car);
+            Socket socket = new Socket("localhost", 4444);
+            ObjectOutputStream os = new ObjectOutputStream(socket.getOutputStream());
```
Comment (kolyansv) [inline producer/src/main/java/controller/NewCarsSender.java]: why do you need to send anything to server at all? Client should only receive cars!

File: producer/src/main/java/controller/NewCarsSender.java [hunk 14]
```diff
@@ -35,13 +36,17 @@ public static void main(String[] args) throws InterruptedException {
         Producer<String, Car> producer = new KafkaProducer<String, Car>(props);
 
         while (true) {
-            //Creating new car
             Car car = new Car();
-            carService.makeCar(car);
+            Socket socket = new Socket("localhost", 4444);
```
Comment (kolyansv) [inline producer/src/main/java/controller/NewCarsSender.java]: DO NOT hardcode values

**Review by kolyansv — CHANGES_REQUESTED** [review]

File: server/src/main/java/server/Server.java (hunk 7 above)
Comment (tarasulo) [reply server/src/main/java/server/Server.java]: delete

File: server/src/main/java/server/Server.java (hunk 8 above)
Comment (tarasulo) [reply server/src/main/java/server/Server.java]: delete, wrong example

File: server/src/main/java/server/Server.java (hunk 10 above)
Comment (tarasulo) [reply server/src/main/java/server/Server.java]: it's from previous version when I've tried start Server from another controller

File: server/src/main/java/server/Server.java (hunk 11 above)
Comment (tarasulo) [reply server/src/main/java/server/Server.java]: no, iI changed it

File: server/src/main/java/server/Server.java (hunk 12 above)
Comment (tarasulo) [reply server/src/main/java/server/Server.java]: Uppercase

File: producer/src/main/java/controller/NewCarsSender.java (hunk 13 above)
Comment (tarasulo) [reply producer/src/main/java/controller/NewCarsSender.java]: I deleted it

File: producer/src/main/java/controller/NewCarsSender.java (hunk 14 above)
Comment (tarasulo) [reply producer/src/main/java/controller/NewCarsSender.java]: ok

File: server/src/main/java/server/Server.java (hunk 9 above)
Comment (tarasulo) [reply server/src/main/java/server/Server.java]: ok

**Pull request merged**